lib/core/src/lib/datatable/components/datatable/datatable.component.spec.ts

```diff
@@ -415,6 +415,24 @@ describe('DataTable', () => {
         expect(dataTable.resetSelection).toHaveBeenCalled();
     });
 
+    it('should reset selection on multiselect change', () => {
```

Comment:
do we care only about multiselect? what if the type is different do we want or not reset? 
Can you please cover other cases to make sure the expectations are the one we expect?

Reply:
I don't think I understand the questions 😅. The reported bug is about rows not being deselected, when multiselect changes to false (when we hide the checkboxes). I've tested my changes on HxP and I haven't noticed any unwanted behavior.

Reply:
Also right above the unit I added is `should reset selection on mode change`